src/SfCommandError.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { SfError, StructuredMessage } from '@salesforce/core';
+import { AnyJson } from '@salesforce/ts-types';
+import { computeErrorCode } from './errorHandling.js';
+import { removeEmpty } from './util.js';
+
+// These types are 90% the same as SfErrorOptions (but they aren't exported to extend)
+type ErrorDataProperties = AnyJson;
+export type SfCommandErrorOptions<T extends ErrorDataProperties = ErrorDataProperties> = {
+  message: string;
+  exitCode: number;
+  code: string;
+  name?: string;
+  commandName: string;
+  data?: T;
+  cause?: unknown;
+  context?: string;
+  actions?: string[];
+  result?: unknown;
+  warnings?: Array<StructuredMessage | string>;
+};
+
+type SfCommandErrorJson = {
+  name: string;
+  message: string;
+  exitCode: number;
+  commandName: string;
+  context: string;
+  code: string;
+  status: string;
+  stack: string;
+  actions?: string;
+  data?: ErrorDataProperties;
+  cause?: string;
+  warnings?: Array<StructuredMessage | string>;
+  result?: unknown;
+};
```

Comment:
suggestion: it looks like there's a shared type that both of these real types could `&` from, so it's easier to see the differences